openprescribing/api/views_spending.py

```diff
@@ -385,266 +384,341 @@ def spending_by_ccg(request, format=None):
-            for i, c in enumerate(codes):
-                query += "pr.chemical_id LIKE %s "
-                if (i != len(codes) - 1):
-                    query += ' OR '
-            codes = [c + '%' for c in codes]
-        else:
-            for i, c in enumerate(codes):
-                query += "pr.chemical_id=%s "
-                if (i != len(codes) - 1):
-                    query += ' OR '
-        query += ") "
-    if orgs:
-        query += "AND ("
-        for i, org in enumerate(orgs):
-            query += "pr.pct_id=%s "
-            if (i != len(orgs) - 1):
-                query += ' OR '
-        query += ") "
-    query += "GROUP BY pc.code, pc.name, date "
-    query += "ORDER BY date, pc.code "
-    return query
-
-
-def _get_query_for_presentations_by_ccg(codes, orgs):
-    query = 'SELECT pc.code as row_id, '
-    query += "pc.name as row_name, "
-    query += 'pr.processing_date as date, '
-    query += "SUM(pr.items) AS items, "
-    query += 'SUM(pr.cost) AS actual_cost, '
-    query += 'SUM(pr.quantity) AS quantity '
-    query += "FROM vw__presentation_summary_by_ccg pr "
-    query += "JOIN frontend_pct pc ON pr.pct_id=pc.code "
-    query += "AND pc.org_type='CCG' "
-    query += " WHERE ("
-    for i, c in enumerate(codes):
-        query += "pr.presentation_code LIKE %s "
-        if (i != len(codes) - 1):
-            query += ' OR '
-    if orgs:
-        query += ") AND ("
-        for i, org in enumerate(orgs):
-            query += "pr.pct_id=%s "
-            if (i != len(orgs) - 1):
-                query += ' OR '
-    query += ") GROUP BY pc.code, pc.name, date "
-    query += "ORDER BY date, pc.code"
-    return query
-
-
-def _get_total_spending_by_practice(orgs, date):
-    query = 'SELECT pr.practice_id AS row_id, '
-    query += "pc.name AS row_name, "
-    query += "pc.setting AS setting, "
-    query += "pc.ccg_id AS ccg, "
-    query += 'pr.processing_date AS date, '
-    query += 'pr.cost AS actual_cost, '
-    query += 'pr.items AS items, '
-    query += 'pr.quantity AS quantity '
-    query += "FROM vw__practice_summary pr "
-    query += "JOIN frontend_practice pc ON pr.practice_id=pc.code "
-    if orgs or date:
-        query += "WHERE "
+        code_clauses = None
+
+    where_condition = _build_where_condition([code_clauses])
+
+    return query % where_condition
+
+
+def _get_query_for_chemicals_or_sections_by_ccg(codes, pct_ids, spending_type):
+    query = '''
+    SELECT pc.code as row_id,
+        pc.name as row_name,
+        pr.processing_date as date,
+        SUM(pr.cost) AS actual_cost,
+        SUM(pr.items) AS items,
+        SUM(pr.quantity) AS quantity
+    FROM vw__chemical_summary_by_ccg pr
+    JOIN frontend_pct pc ON pr.pct_id=pc.code
+    AND pc.org_type='CCG'
+    WHERE %s
+    GROUP BY pc.code, pc.name, date
+    ORDER BY date, pc.code
+    '''
+
+    if spending_type == 'bnf-section':
+        chemical_clauses = [
+            'pr.chemical_id LIKE %s'
```

Comment:
Took me a minute to parse this in my head. Might `['pr.chemical_id LIKE %s'] * len(codes)` be a tad clearer?